packages/app/src/components-styled/stacked-chart/stacked-chart.tsx

```diff
@@ -332,36 +332,39 @@ export function StackedChart<T extends Value>(props: StackedChartProps<T>) {
    *
    * @TODO wrap in useCallback?
    */
-  function handleHover(
-    event: HoverEvent,
-    tooltipData: TooltipData,
-    hoverIndex: number
-  ) {
-    const isLeave = event.type === 'mouseleave';
-
-    if (isLeave) {
-      tooltipTimeout = window.setTimeout(() => {
-        hideTooltip();
-      }, 300);
-      hoverTimeout = window.setTimeout(() => {
-        setHoveredIndex(NO_HOVER_INDEX);
-      }, 300);
-      return;
-    }
-
-    if (tooltipTimeout) clearTimeout(tooltipTimeout);
-    if (hoverTimeout) clearTimeout(hoverTimeout);
-
-    setHoveredIndex(hoverIndex);
-
-    // @ts-expect-error
-    const coords = localPoint(event.target.ownerSVGElement, event);
-    showTooltip({
-      tooltipLeft: coords?.x || 0,
-      tooltipTop: coords?.y || 0,
-      tooltipData,
-    });
-  }
+  const handleHover = useCallback(
+    function handleHover(
+      event: HoverEvent,
+      tooltipData: TooltipData,
+      hoverIndex: number
+    ) {
+      const isLeave = event.type === 'mouseleave';
+
+      if (isLeave) {
+        tooltipTimeout = window.setTimeout(() => {
+          hideTooltip();
+        }, 300);
+        hoverTimeout = window.setTimeout(() => {
+          setHoveredIndex(NO_HOVER_INDEX);
+        }, 300);
```

Comment:
it's safer to validate the component is still mounted when the timeout ends (especially with these quite long durations), otherwise we'll have runtime errors due to mutating state of an unmounted component.

You could create this hook:
```tsx
export function useIsMountedRef() {
  const isMountedRef = useRef(false);

  useEffect(() => {
    isMountedRef.current = true;
    return () => {
      isMountedRef.current = false;
    };
  }, []);

  return isMountedRef;
}
```

And use it like this:
```suggestion
        tooltipTimeout = window.setTimeout(() => {
          if (isMountedRef.current) hideTooltip();
        }, 300);
        hoverTimeout = window.setTimeout(() => {
          if (isMountedRef.current) setHoveredIndex(NO_HOVER_INDEX);
        }, 300);
```